internal/eventstore/models/aggregate.go

```diff
@@ -15,15 +15,21 @@ func (at AggregateType) String() string {
 type Aggregates []*Aggregate
 
 type Aggregate struct {
-	id             string
-	typ            AggregateType
-	latestSequence uint64
-	version        Version
+	id               string
+	typ              AggregateType
+	PreviousSequence uint64
+	version          Version
 
 	editorService string
 	editorUser    string
 	resourceOwner string
 	Events        []*Event
+	Precondition  *precondition
+}
+
+type precondition struct {
+	Query        *SearchQuery
+	Precondition func(...*Event) error
```

Comment:
we could name it Validation / Validate as you did in the SetPrecondition method... so we don't have Precondition.Precondition internal/eventstore/internal/repository/sql/push.go

Reply:
renamed